react-app/src/modules/account/Account.js

```diff
@@ -0,0 +1,120 @@
+import scrypt from 'scrypt-async';
+import nacl from 'tweetnacl';
+import sha256 from 'fast-sha256';
+import blake from 'blakejs';
+
+nacl.util = require('tweetnacl-util');
+
+const OFFSET_MIN = -12;
+const OFFSET_MAX = 14;
+
+class Account {
+  constructor() {
+    this._initialized = false;
+  }
+
+  initialize(username, password, offset) {
```

Comment:
Why have an initialize and check for this everywhere instead of just having `constructor(username, password)`?

Reply:
The current implementation exports an instance of the Account class, not the class itself, so we have a singleton-like implementation that should be initialized while signin. However, it may be neccessary to allow the possibility to re-initialize the instance to support signout/signin actions-pair.

@ivanalejandro0, is this right what I'm saying?

Reply:
Well, if you export the instance, why do you need the singleton? A singleton is needed when you export the **class**, so you actively prevent more than one instance in your application.

If you export an object only, then there's no way to create a new one anyway.

Reply:
Then I would remove all the initialize logic, treat it as a regular class, just dont export anything but the instance. It'll simplify the code.